tree/ntuple/v7/inc/ROOT/RNTupleProcessor.hxx

```diff
@@ -0,0 +1,293 @@
+   };
+
+   // clang-format off
+   /**
+   \class ROOT::Experimental::RNTupleProcessor::RProcessorView
+   \ingroup NTuple
+   \brief View on the RNTupleProcessor iterator state.
+   */
+   // clang-format on
+   class RProcessorView {
+   private:
+      const REntry &fEntry;
+      const NTupleSize_t &fGlobalEntryIndex;
+      const NTupleSize_t &fLocalEntryIndex;
+      const std::size_t &fNTupleIndex;
+
+   public:
+      RProcessorView(const REntry &entry, const NTupleSize_t &globalEntryIndex, const NTupleSize_t &localEntryIndex,
+                     const std::size_t &ntupleIndex)
+         : fEntry(entry),
+           fGlobalEntryIndex(globalEntryIndex),
+           fLocalEntryIndex(localEntryIndex),
+           fNTupleIndex(ntupleIndex)
+      {
+      }
+
+      const REntry *operator->() const { return &fEntry; }
+      const REntry &GetEntry() const { return fEntry; }
+      NTupleSize_t GetGlobalEntryIndex() const { return fGlobalEntryIndex; }
+      NTupleSize_t GetLocalEntryIndex() const { return fLocalEntryIndex; }
+      std::size_t GetNTupleIndex() const { return fNTupleIndex; }
+   };
+
+   const std::vector<RNTupleSourceSpec> &fNTuples;
+   std::unique_ptr<RNTupleModel> fModel;
+   std::unique_ptr<REntry> fEntry;
+   std::unique_ptr<Internal::RPageSource> fPageSource;
+   std::vector<RFieldContext> fFieldContexts;
+
+   /////////////////////////////////////////////////////////////////////////////
+   /// \brief Connect an RNTuple for processing.
+   ///
+   /// \param[in] ntuple The RNTupleSourceSpec describing the RNTuple to connect.
+   ///
+   /// \return The number of entries in the newly-connected RNTuple.
+   ///
+   /// Creates and attaches new page source for the specified RNTuple, and connects the fields that are known by
+   /// the processor to it.
+   NTupleSize_t ConnectNTuple(const RNTupleSourceSpec &ntuple);
+
+   /////////////////////////////////////////////////////////////////////////////
+   /// \brief Creates and connects concrete fields to the current page source, based on the proto-fields.
+   void ConnectFields();
+
+   /////////////////////////////////////////////////////////////////////////////
+   /// \brief Loads an entry.
+   ///
+   /// \param[in] index The index (local to the current RNTuple) of the entry to load.
+   void LoadEntry(NTupleSize_t index) { fEntry->Read(index); }
+
+public:
+   /////////////////////////////////////////////////////////////////////////////
+   /// \brief Returns a reference to the entry used by the processor.
+   ///
+   /// \return A reference to the entry used by the processor.
+   ///
+   const REntry &GetEntry() const { return *fEntry; }
+
+   /////////////////////////////////////////////////////////////////////////////
+   /// \brief Returns a reference to the page source currently used in the processor.
+   ///
+   /// \return A reference to the page source.
+   Internal::RPageSource &GetPageSource() const { return *fPageSource; }
```

Comment:
Here we are exposing an internal class through a public API, something to consider for modification